apps/recovery-portal/src/main/webapp/self-registration-complete.jsp

```diff
@@ -85,6 +135,28 @@
             <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "Close")%>
         </button>
     </div>
+    <form id="callbackForm" name="callbackForm" method="post" action="/commonauth">
+        <%
+            if (username != null) {
+        %>
+        <div>
+            <input type="hidden" name="username"
+                   value="<%=Encode.forHtmlAttribute(username)%>"/>
```

Comment:
Can't have this in the previous line itself?

Reply:
yes sure